.azure/pipelines/ci.yml

```
@@ -471,3 +471,50 @@ jobs:
     - name: Linux_Test_Results
       path: artifacts/TestResults/
       publishOnError: true
+
+# Source build
+- job: Source_Build
+  displayName: 'Test: Linux Source Build'
+  container: centos:7
+  pool:
+    vmImage: 'ubuntu-16.04'
+  variables:
+    DotNetCoreSdkDir: $(Agent.ToolsDirectory)/dotnet
+    DOTNET_SYSTEM_GLOBALIZATION_INVARIANT: true
+  steps:
+  - script: |
+      source eng/common/native/common-library.sh
+      mkdir -p $HOME/bin
+      GetFile https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 $HOME/bin/jq
+      chmod +x $HOME/bin/jq
+      echo "##vso[task.prependpath]$HOME/bin"
+    displayName: Install jq
+  - task: UseDotNet@2
+    displayName: 'Use .NET Core sdk'
+    inputs:
+      packageType: sdk
+      # The SDK version selected here is intentionally supposed to use the latest release
+      # For the purpose of building Linux distros, we can't depend on features of the SDK
+      # which may not exist in pre-built versions of the SDK
+      version: 3.0.x
+      installationPath: $(DotNetCoreSdkDir)
+      includePreviewVersions: true
+  - script: ./eng/scripts/ci-source-build.sh --ci --configuration Release /p:BuildManaged=true
+    displayName: Run ci-source-build.sh
+  - task: PublishBuildArtifacts@1
+    displayName: Upload logs
+    condition: always()
+    continueOnError: true
+    inputs:
+      pathtoPublish: artifacts/log/
+      artifactName: Source_Build_Logs
+      artifactType: Container
+      parallel: true
+  - task: PublishBuildArtifacts@1
+    displayName: Upload package artifacts
+    condition: always()
```

Never upload packages for builds from a fork. See for example the equivalent in Extensions:
```suggestion
    # Only capture source build artifacts in PRs for the sake of inspecting
    # changes that impact source-build. The artifacts from this build pipeline are never actually used.
    condition: and(succeeded(), eq(variables['Build.Reason'], 'PullRequest'))
```

---

Actually, how about I set it to just Build.Reason = PullRequest. I was debugging earlier and the succeed() condition meant I couldn't get the artifacts.

---

That's fine while you're debugging but not once this is checked in. Look at all other artifacts published in this pipeline. Only artifacts/log files are ever published on error.

---

Oh, you've left `succeeded()` in. Alright&hellip;

---

I made the comment after merging your suggestion, wanted to see what you think before I make the changes.